docs/php-api.md

```diff
@@ -0,0 +1,734 @@
+  - If the input does not validate against the input schema, the permission callback will not be called, and a `WP_Error` will be returned instead.
+- `meta` (`array`, **Optional**): An associative array for storing arbitrary additional metadata about the ability.
+  - `annotations` (`array`, **Optional**): An associative array of annotations providing hints about the ability's behavior characteristics. Supports the following keys:
+    - `instructions` (`string`, **Optional**): Custom instructions or guidance for using the ability (default: `''`).
+    - `readonly` (`boolean`, **Optional**): Whether the ability only reads data without modifying its environment (default: `false`).
+    - `destructive` (`boolean`, **Optional**): Whether the ability may perform destructive updates to its environment. If `true`, the ability may perform any type of modification, including deletions or other destructive changes. If `false`, the ability performs only additive updates (default: `true`).
+    - `idempotent` (`boolean`, **Optional**): Whether calling the ability repeatedly with the same arguments will have no additional effect on its environment (default: `false`).
+  - `show_in_rest` (`boolean`, **Optional**): Whether to expose this ability via the REST API. Default: `false`.
+    - When `true`, the ability will be listed in REST API responses and can be executed via REST endpoints.
+    - When `false`, the ability will be hidden from REST API listings and cannot be executed via REST endpoints, but remains available for internal PHP usage.
+- `ability_class` (`string`, **Optional**): The fully-qualified class name of a custom ability class that extends `WP_Ability`. This allows you to customize the behavior of an ability by extending the base `WP_Ability` class and overriding its methods. The custom class must extend `WP_Ability`. Default: `WP_Ability`.
+
+### Ability ID Convention
+
+The `$id` parameter must follow the pattern `namespace/ability-name`:
+
+- **Format:** Must contain only lowercase alphanumeric characters (`a-z`, `0-9`), hyphens (`-`), and one forward slash (`/`) for namespacing.
+- **Convention:** Use your plugin slug as the namespace, like `my-plugin/ability-name`.
+- **Examples:** `my-plugin/update-settings`, `woocommerce/get-product`, `contact-form/send-message`, `analytics/track-event`
+
+### Code Examples
+
+#### Registering a Simple Data Retrieval Ability
+
+```php
+add_action( 'abilities_api_init', 'my_plugin_register_site_info_ability' );
+function my_plugin_register_site_info_ability() {
+    wp_register_ability( 'my-plugin/get-site-info', array(
+        'label' => __( 'Get Site Information', 'my-plugin' ),
+        'description' => __( 'Retrieves basic information about the WordPress site including name, description, and URL.', 'my-plugin' ),
+        'category' => 'data-retrieval',
+        'input_schema' => array(
```

Comment:
If there so need for input the schema can be omitted and callbacks will not receive the input param. It would be great to illustrate that here.